File: service-backbase-java/src/main/java/org/example/backend/model/response/ArticleDetailsResponseBody.java
package org.example.backend.model.response;

import java.util.List;
import pl.net.testit.serum.commons.json.JsonEntity;

public class ArticleDetailsResponseBody extends JsonEntity {

  public Article article;

  public static class Article {

    public String slug;
    public String title;
    public String description;
    public String body;
    public List<String> tagList;
    public String createdAt;
    public String updatedAt;
    public Boolean favorited;
    public Integer favoritesCount;
    public Author author;

  }

  public static class Author {

    public String username;
    public String bio;
    public String image;
    public Boolean following;
  }
}

File: service-backbase-kotlin/src/main/kotlin/org/example/backend/model/response/ArticleDetailsResponseBody.kt
package org.example.backend.model.response

class ArticleDetailsResponseBody(
  val article: ArticleDetailsResponseBodyArticle? = null,
)

class ArticleDetailsResponseBodyArticle(
  val slug: String? = null,
  val title: String? = null,
  val description: String? = null,
  val body: String? = null,
  val tagList: List<String>? = null,
  val createdAt: String? = null,
  val updatedAt: String? = null,
  val favorited: Boolean? = null,
  val favoritesCount: Int? = null,
  val author: ArticleDetailsResponseBodyArticleAuthor? = null,
)

class ArticleDetailsResponseBodyArticleAuthor(
  val username: String? = null,
  val bio: String? = null,
  val image: String? = null,
  val following: Boolean? = null,
)